services/settings-service/src/main/java/com/amazon/aws/partners/saasfactory/saasboost/SettingsServiceDAL.java

```diff
@@ -647,7 +714,10 @@ public static Parameter toTenantParameterStore(UUID tenantId, Setting setting) {
         settings.add(Setting.builder().name("MIN_COUNT").value(minCount).readOnly(false).build());
         String maxCount = appConfig.getMaxCount() != null ? appConfig.getMaxCount().toString() : null;
         settings.add(Setting.builder().name("MAX_COUNT").value(maxCount).readOnly(false).build());
-        settings.add(Setting.builder().name("CLUSTER_OS").value(appConfig.getOperatingSystem().name()).readOnly(false).build());
+        OperatingSystem os = appConfig.getOperatingSystem();
+        if (os != null) {
+            settings.add(Setting.builder().name("CLUSTER_OS").value(os.name()).readOnly(false).build());
+        }
```

Comment:
This will cause a null pointer in [tenant onboarding](https://github.com/awslabs/aws-saas-boost/blob/main/services/onboarding-service/src/main/java/com/amazon/aws/partners/saasfactory/saasboost/OnboardingService.java#L765) if the `CLUSTER_OS` parameter isn't included.